caas/kubernetes/provider/detectcloud.go

```diff
@@ -54,7 +54,13 @@ func (p kubernetesEnvironProvider) DetectCloud(name string) (cloud.Cloud, error)
 	if err == nil && name == k8s.K8sCloudMicrok8s {
 		return mk8sCloud, nil
 	}
+
 	if !errors.IsNotFound(err) && err != nil {
+		// if the cloud is not microk8s and we get the user group error, return not found so the caller skips the cloud
```

Comment:
Can we check the error content to make sure it is the user group error. We probably want to still raise an error if something else goes wrong.

Reply:
Sure thing.

Reply:
EDIT: The only time the error from the microk8s provider matters, is if we are bootstrapping a microk8s cloud. Else we return a notfound regardless of the error so that the caller know to skip the cloud.